File: doc-framework/test/templates/realtime.tpl.java
import io.kuzzle.sdk.*;
import java.util.Date;
import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class CodeExampleGenericClass {
    private static Kuzzle kuzzle;

    public static void main(String[] args) {
      WebSocket ws = new WebSocket("kuzzle");
      kuzzle = new Kuzzle(ws);
      kuzzle.connect();

      ByteArrayOutputStream baos = new ByteArrayOutputStream();
      PrintStream ps = new PrintStream(baos);
      PrintStream outOriginal = System.out;

      System.setOut(ps);

      [snippet-code]

      for (int i = 150; i > 0 && baos.size() == 0; --i) {
        try {
          Thread.sleep(200);
        } catch (InterruptedException e) {
          System.err.println(e.getMessage());
        }
      }

      System.out.flush();
      System.setOut(outOriginal);
      System.out.println(baos.toString());
    }
}

File: doc-framework/test/templates/realtime.tpl.go
package main

import (
	"fmt"
	"log"
	"os"
	"time"
	"encoding/json"

	"github.com/kuzzleio/sdk-go/protocol/websocket"
	kuzzlepkg "github.com/kuzzleio/sdk-go/kuzzle"
	"github.com/kuzzleio/sdk-go/types"
)

func main() {
	c := websocket.NewWebSocket("kuzzle", nil)
	kuzzle, _ := kuzzlepkg.NewKuzzle(c, nil)

	connectErr := kuzzle.Connect()
	if connectErr != nil {
		log.Fatal(connectErr)
		os.Exit(1)
	}

	rescueStdout := os.Stdout
	r, w, _ := os.Pipe()
	os.Stdout = w

	[snippet-code]

	go func() {
		time.Sleep(30 * time.Second)
		os.Exit(1)
	}()

	b := make([]byte, 4096)
	n, _ := r.Read(b)

	os.Stdout = rescueStdout
	fmt.Print(string(b[:n]))
}
